exporter/otlp/test/opentelemetry/exporter/otlp/exporter_test.rb

```diff
@@ -182,6 +182,15 @@
       _(result).must_equal(FAILURE)
     end
 
+    it 'returns FAILURE when encryption to receiver endpoint fails' do
+      stub_request(:post, 'https://localhost:4317/v1/traces').to_raise(OpenSSL::SSL::SSLError.new('enigma wedged'))
+      span_data = create_span_data
+      tight_retry_interval = ->(_) { 0.01 }
```

Comment:
This is dirty and I do feel bad about it. I tried using [the prior art](https://github.com/open-telemetry/opentelemetry-ruby/blob/980f0e454c59e822de439635ee513e2bcdf820b8/exporter/otlp/test/opentelemetry/exporter/otlp/exporter_test.rb#L166-L172) for stubbing `backoff?`, but it had some weird interactions that prevented `send_bytes()` from ever reaching the last retry attempt for the `rescue` blocks to get past `backoff?` and return a `FAILURE`. I suppose another alternative to this would be to stub `backoff?` to return false to short-circuit retries.

Reply:
Without doing something about `backoff?`, the default number of retries (5), the default overall timeout (10 seconds), and `backoff?'s exponentially-growing random sleeping fails attempts to verify that `FAILURE` is returned by any of the exception handling. Their overall timeout time is exceeded before the 5 retries complete.

Reply:
Did you include the previous line `@retry_count = 0` when stubbing `backoff?` (and the `ensure` block as well, though that’s probably overkill)?

I think stubbing `backoff?` to return `false` is the right approach here. You shouldn’t need to care about retries for this test.

Reply:
Self shunting is perfectly reasonable in my opinion. 

Other options maybe to stub the `RETRY` constant or allow for the retries to be configurable and default to the constant value. 

I personally lean towards making the exporter configurable because it encapsulates the functionality in the exporter but do not object to using this testing pattern.

Reply:
> Did you include the previous line @retry_count = 0 when stubbing backoff?

Pretty sure I did, but those buffers are lost to me now.

![poof](https://user-images.githubusercontent.com/517302/126413680-8eaa76f1-d4fe-4f6f-9942-063cc24f47aa.gif)

> I think stubbing `backoff?` to return `false` is the right approach here. You shouldn’t need to care about retries for this test.

I agree that's better. Or at least less bad. Though, I'll note that with the default exporter configuration, `send_bytes()` will rarely if ever hit the `return FAILURE`s in the rescue blocks. `send_bytes()` will [`return TIMEOUT`](https://github.com/open-telemetry/opentelemetry-ruby/blob/980f0e454c59e822de439635ee513e2bcdf820b8/exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb#L167) as the 10 seconds is exceeded in the first 3-4 attempts. We'll be relying on deltas in the `otel.otlp_exporter.failure` metrics emitted to track down gremlins within the timeouts.